core/src/test/scala/org/bykn/bosatsu/LetFreeTest.scala

```diff
@@ -56,14 +68,87 @@ package Recur/Some
 def foo(x):
   recur x:
     []: ["a","b","c"]
+    ["a: ${bar}", *t]: NonEmptyList(bar, foo(t))
     [_, *t]: NonEmptyList("zero", foo(t))
 
 out = foo
-"""
-      ), "Recur/Some", Recursion(Lambda(Lambda(Match(LambdaVar(0),NonEmptyList.of(
-        (PositionalStruct(Some(0), Nil),Struct(1,List(Literal(Str("a")), Struct(1,List(Literal(Str("b")), Struct(1,List(Literal(Str("c")), Struct(0,List())))))))),
-      (PositionalStruct(Some(1), List(WildCard, Var(0))),Lambda(Struct(1,List(Literal(Str("zero")), App(LambdaVar(2),LambdaVar(0))))))
-      )))))
+"""),
+      "Recur/Some",
+      Recursion(
```

Comment:
I get really worried about these tests. In the past, minor changes elsewhere in the code have changed these results, which can then only be updated by very carefully building the AST.

An alternate idea for such gold-set tests would be to assert that the toString or asString or whatever hashes to a given value. Then to update the test you only need to update the hash. When the hash fails, you can print what the original AST is and leave it in the comment as an expectation for someone to update, but the update process becomes much easier.

Also, is this a content change or just a formatting change? If it is a formatting change, can you make it a separate PR?

Reply:
They aren't particularly great tests because the expressions are so big to be totally opaque. They are usually here to test some edge of normalization.

Let me see if I can make something better.

I believe in this case there was a bug and the expression did change in some subtle way.